packages/grid/src/vaadin-grid-row-state-mixin.js

```diff
@@ -0,0 +1,64 @@
+/**
+ * @license
+ * Copyright (c) 2016 - 2022 Vaadin Ltd.
+ * This program is available under Apache License Version 2.0, available at https://vaadin.com/license/
+ */
+
+/**
+ * @polymerMixin
+ */
+export const RowStateMixin = (superClass) =>
+  class RowStateMixin extends superClass {
+    static get observers() {
+      return ['__updateHeaderFooterRows(size, _columnTree, _columnTree.*)'];
+    }
+
+    /** @private */
+    __iterateRows(container, callback) {
+      [...container.children].forEach(callback);
+    }
```

Comment:
This method might be worth extracting to `vaadin-grid-helper.js` as we use this pattern in grid quite a lot. Created a separate PR to keep the diff in this one smaller, please check it out: #5059